File: app/src/main/java/com/example/smartcityassistant/data/ai/AiModels.kt
package com.example.smartcityassistant.data.ai

data class AiMessage(
    val id: String,
    val text: String,
    val isUser: Boolean,
    val timestamp: Long = System.currentTimeMillis(),
    val actionCard: AiActionCardData? = null
)

data class AiActionCardData(
    val title: String,
    val description: String,
    val actionText: String = "Open",
    val intent: String? = null,
    val action: String? = null,
    val parameters: Map<String, String>? = null
)

data class AiResponse(
    val reply: String,
    val intent: String? = null,
    val action: String? = null,
    val parameters: Map<String, String>? = null,
    val actionCard: AiActionCardData? = null
)

File: railway-backend/src/models/aiModels.ts
export interface AiMessageRequest {
  message: string;
}

export interface AiActionCardData {
  title: string;
  description: string;
  actionText?: string;
  intent?: string;
  action?: string;
  parameters?: Record<string, string>;
}

export interface AiResponseData {
  reply: string;
  intent: string;
  action: string | null;
  parameters: Record<string, any>;
  actionCard?: AiActionCardData;
}

export interface AiApiResponse {
  success: boolean;
  data?: AiResponseData;
  error?: {
    code: string;
    message: string;
  };
}
